src/Sarif.Viewer.VisualStudio/Models/SarifErrorListItem.cs

```diff
@@ -188,16 +204,23 @@ public SarifErrorListItem(Run run, Notification notification, string logFilePath
 
             Level = notification.Level;
             LogFilePath = logFilePath;
-            FileName = SdkUIUtilities.GetFileLocationPath(notification.Locations?[0]?.PhysicalLocation?.ArtifactLocation, _runId) ?? string.Empty;
+            FileName = SdkUIUtilities.GetFileLocationPath(notification.Locations?[0]?.PhysicalLocation?.ArtifactLocation, RunIndex) ?? string.Empty;
             ProjectName = projectNameCache.GetName(FileName);
-            Locations.Add(new LocationModel() { FilePath = FileName });
+            Locations.Add(new LocationModel(this.ResultId, this.RunIndex) { FilePath = FileName });
 
             Tool = run.Tool.ToToolModel();
             Rule = rule.ToRuleModel(ruleId);
             Invocation = run.Invocations?[0]?.ToInvocationModel();
-            WorkingDirectory = Path.Combine(Path.GetTempPath(), _runId.ToString());
+            WorkingDirectory = Path.Combine(Path.GetTempPath(), RunIndex.ToString());
         }
 
+        /// <summary>
+        /// Gets the result ID that uniquely identifies this result for this Visual Studio session.
+        /// </summary>
+        public int ResultId { get; }
+
+        private int RunIndex { get; }
+
```

Comment:
Verify why one of this is public, and one is private and then comment why this is.